### PR TITLE
adding optional memcache container to rucio did finder pod

### PR DESCRIPTION
**Pull request by ivukotic**
this is needed to address issue: https://github.com/ssl-hep/ServiceX_DID_Finder_Rucio/issues/52

### CODE REVIEW

File: servicex/templates/did-finder-rucio/deployment.yaml [hunk 1]
```diff
@@ -23,38 +23,46 @@ spec:
...
+          {{- end }}
         volumeMounts:
           - name: rucio-cfg
             mountPath: /opt/rucio/etc/
           - name: x509-secret
             mountPath: /etc/grid-security-ro
             readOnly: true
+      {{- if ((.Values.didFinder.rucio.memcache).enabled) }}
+      - name: {{ .Release.Name }}-did-finder-rucio-memcache
+        image: memcached:{{ .Values.didFinder.rucio.memcache.tag }}
```
Comment (BenGalewsky) [inline servicex/templates/did-finder-rucio/deployment.yaml]: Burt has requested that we don't have hardcoded docker images in our charts - he will want to use a local copy of this image so override it in `values.yaml`

**Comment by BenGalewsky** [comment]
Do you really think memcache is needed here? It seems that there are simpler pure python implementations possible.

File: servicex/templates/did-finder-rucio/deployment.yaml (hunk 1 above)
Comment (ivukotic) [reply servicex/templates/did-finder-rucio/deployment.yaml]: OK I understand.

**Review by BenGalewsky — CHANGES_REQUESTED** [review]
Please update the reference documentation `docs/deployment/reference.md`

**Comment by ivukotic** [comment]
> Please update the reference documentation `docs/deployment/reference.md`

done

**Review by BenGalewsky — APPROVED** [review]

**Pull request merged**